### PR TITLE
Improve error handling around the existence (or not) of a config file

### PR DESCRIPTION
**Pull request by rpowis**
* Remove the hard dependency on the existence of a `config.json` file for tests (gets travis to pass)
* Log a human readable error when trying to run without a `config.json` file (UX improvement)

### CODE REVIEW

File: finder.js [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+import {EOL} from 'os';
 import clone from './lib/clone';
 import search from './lib/search';
 import serviceCatalogue from './lib/serviceCatalogue';
 import isFrontendService from './lib/utils/isFrontendService';
 
+try {
+  var config = require('./config.json');
+}
+catch(err) {
```
Comment (feedmypixel) [inline finder.js]: this can go on the line above (this project ideally should have js linting)

File: finder.js [hunk 2]
```diff
@@ -1,9 +1,18 @@
-import config from './config';
+import fs from 'fs';
+import {EOL} from 'os';
 import clone from './lib/clone';
 import search from './lib/search';
 import serviceCatalogue from './lib/serviceCatalogue';
 import isFrontendService from './lib/utils/isFrontendService';
 
+try {
+  var config = require('./config.json');
```
Comment (feedmypixel) [inline finder.js]: const please

File: finder.js [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
+}
+catch(err) {
+  process.stdout.write(`No config.json file found ${EOL}`);
```
Comment (feedmypixel) [inline finder.js]: can we throw here rather than logging?

File: test/serviceCatalogue.test.js [hunk 4]
```diff
@@ -40,6 +49,25 @@ const mockLibraries = [
   }
 ];
 
+test('.getProjects() should barf without config', t => {
```
Comment (feedmypixel) [inline test/serviceCatalogue.test.js]: "should throw"

File: test/serviceCatalogue.test.js [hunk 5]
```diff
@@ -40,6 +49,25 @@ const mockLibraries = [
...
+  t.throws(() => serviceCatalogue.getProjects(), 'No config object given');
+});
+
+test('.getProjects() should return an array of projects', async t => {
+  nock(`${apiConfig.protocol}://${apiConfig.host}:443`)
+    .get(servicesAPIPath)
+    .reply(200, mockServices);
+
+  nock(`${apiConfig.protocol}://${apiConfig.host}:443`)
```
Comment (feedmypixel) [inline test/serviceCatalogue.test.js]: this could go in a `const` as its used more than once in this file

File: test/serviceCatalogue.test.js [hunk 6]
```diff
@@ -40,6 +49,25 @@ const mockLibraries = [
...
+    .get(librariesAPIPath)
+    .reply(200, mockLibraries);
+
+  const projects = await serviceCatalogue.getProjects(apiConfig);
+
+  t.true(Array.isArray(projects));
+  t.is(projects.length, mockServices.length + mockLibraries.length);
```
Comment (feedmypixel) [inline test/serviceCatalogue.test.js]: you could add `t.deepEqual(projects, mockServices.concat(mockLibraries));`

File: finder.js (hunk 3 above)
Comment (rpowis) [inline finder.js]: I don't think it's necessary. We're only reporting a missing file and it's not in function or anything. That makes it a UX sugar thing and throwing an error doesn't log to the console nicely.

File: finder.js (hunk 1 above)
Comment (rpowis) [inline finder.js]: That's just personal preference. Let's leave it to the linter.

File: test/serviceCatalogue.test.js (hunk 4 above)
Comment (rpowis) [inline test/serviceCatalogue.test.js]: So sensible. Haha

File: test/serviceCatalogue.test.js (hunk 5 above)
Comment (rpowis) [inline test/serviceCatalogue.test.js]: Sure, but that's got nothing to do with this PR

File: test/serviceCatalogue.test.js (hunk 6 above)
Comment (rpowis) [inline test/serviceCatalogue.test.js]: Again, nothing to do with this PR

**Comment by feedmypixel** [comment]
LGTM 👍

**Pull request merged**